pkgs/build-support/buildenv/default.nix

```diff
@@ -88,30 +88,37 @@ lib.makeOverridable (
       lib.flatten
       (lib.remove null)
     ];
+
+    pkgs = builtins.toJSON chosenOutputs;
   in
   runCommand name
     (
-      rec {
+      {
+        env = {
+          inherit
+            pkgs
+            extraPrefix
+            ignoreCollisions
+            checkCollisionContents
+            ignoreSingleFileOutputs
+            manifest
+            ;
+          pathsToLinkJSON = builtins.toJSON pathsToLink;
+          extraPathsFrom = toString (lib.optional includeClosures (writeClosure pathsForClosure));
+        };
```

Comment:
From PR  #432957's perspective, referencing from `finalAttrs.env.ignoreCollisions` is much less intuitive compared to `finalAttrs.ignoreCollisions`.

In addition, the custom `override` overrider is prone to be overridden by `callPackage`-provided `<pkg>.override` when defining a package, and should be replaced by `<pkg>.overrideAttrs`. Again, overriding `env.ignoreCollisions` is less intuitive than overriding `ignoreCollisions`.

Perl has powerful built-in JSON-parsing support, which we are using to parse `chosenOutputs` (currently stringified as the `pkgs` attribute). Why not parsing `$NIX_ATTRS_JSON_FILE`?